swap/swap_test.go

```diff
@@ -744,6 +745,14 @@ func TestDebtCheques(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+
+	// ...on which we wait until the cashCheque is actually terminated (ensures proper nonce count)
+	select {
+	case <-testBackend.cashDone:
+		log.Debug("cash transaction completed and committed")
+	case <-time.After(4 * time.Second):
+		t.Fatalf("Timeout waiting for cash transactions to complete")
+	}
```

Comment:
Wouldn't it make sense to refactor this in a separate helper function? That way, it will be easier to synchronize these things in tests we make in the future and iyou avoid some code-duplication already now (lines 677-683 are the same).

Reply:
this logic will change soon anyway, so I would leave it for now.